File: app/src/main/java/com/urban/mvvmshowcase/android/MvvmApplication.java
package com.urban.mvvmshowcase.android;

import android.app.Application;
import android.content.Context;
import android.support.annotation.NonNull;

import com.urban.mvvmshowcase.model.repository.FakePersonRepository;
import com.urban.mvvmshowcase.model.repository.PersonRepository;

public class MvvmApplication extends Application {
    private final PersonRepository personRepository = FakePersonRepository.create();

    public static MvvmApplication get(@NonNull Context context) {
        return (MvvmApplication) context.getApplicationContext();
    }

    public PersonRepository getPersonRepository() {
        return personRepository;
    }
}

File: app/src/main/java/com/urban/mvvmshowcase/android/MvvmApplication.kt
package com.urban.mvvmshowcase.android

import android.app.Application
import android.content.Context
import com.urban.mvvmshowcase.person.FakePersonRepository
import com.urban.mvvmshowcase.person.PersonRepository

class MvvmApplication : Application() {
    val personRepository: PersonRepository = FakePersonRepository

    companion object {
        @JvmStatic
        fun get(context: Context): MvvmApplication = context.applicationContext as MvvmApplication
    }
}